File: RpnCalculator-Scala/src/rpn/scala/operators/AdditionOperator.scala
package rpn.scala.operators

import rpn.scala.RpnStack

class AdditionOperator extends Operator {
  val PLUS = "+";

  def performOperation(numbers: RpnStack): Int = {
    val rhs = numbers.pop
    val lhs = numbers.pop
    val result = lhs + rhs
    numbers.push(result)
    return result
  }

  def handlesOperatorCharacter(operand: String): Boolean = {
    operand == PLUS;
  }

}

File: src/main/java/rpn/operator/AdditionOperator.java
package rpn.operator;

public class AdditionOperator extends BinaryOperator {

    private static final String PLUS = "+";

    @Override
    protected int doOperation(int lhs, int rhs) {
        return lhs + rhs;
    }

    @Override
    public boolean handlesOperatorCharacter(String operand) {
        return PLUS.contentEquals(operand);
    }

}
